pytorch_toolkit/ote/ote/modules/arg_converters/base.py

```diff
@@ -15,97 +15,183 @@
 """
 
 import os
-from abc import ABCMeta
-
-
-class BaseArgConverter(metaclass=ABCMeta):
-    train_update_args_map = {}
-    test_update_args_map = {}
-    compress_update_args_map = {}
-    train_to_compress_update_args_map = {}
-    # TODO(LeonidBeynenson): replace dicts train_update_args_map, test_update_args_map,
-    #       and compress_update_args_map with call of a special function
-    #       that may be passed to constructor of the class as a
-    #       parameter.
-    #       This will allow to avoid copying between these dicts.
-
-    train_out_args_map = {
-        'gpu_num': 'gpu_num',
-        'tensorboard_dir': 'tensorboard_dir'
-    }
-    compress_out_args_map = {
-        'gpu_num': 'gpu_num',
-        'tensorboard_dir': 'tensorboard_dir',
-        'nncf_quantization': 'nncf_quantization',
-        'nncf_pruning': 'nncf_pruning',
-        'nncf_sparsity': 'nncf_sparsity',
-        'nncf_binarization': 'nncf_binarization',
-    }
-    test_out_args_map = {
-        'load_weights': 'snapshot',
-        'save_metrics_to': 'out',
-        'save_output_to': 'show_dir',
-    }
-
-    def __init__(self):
+from abc import ABCMeta, abstractmethod
+from collections import namedtuple
+
+
+def map_args(src_args, args_map):
+    return {v: src_args[k] for k, v in args_map.items()}
+
+class ArgConverterMaps(metaclass=ABCMeta):
+    @abstractmethod
+    def train_update_args_map(self):
+        """
+        Returns a map:
+            from: the name of cmd line arguments of the corresponding **train** argparser.parse_args
+                  (e.g. `argparser = DefaultArgParser.get_train_parser()`)
+            to:   the fields of mmdetection/mmaction/other config for its **train** script in the form
+                  compatible with mmcv.Config.merge_from_dict
```

Comment:
mmcv.Config is not always config for any algo

Reply:
Wrote explanation